### PR TITLE
feat(Dropdown): adds popper open support in uncontrolled dropdown

### PR DESCRIPTION
**Pull request by riyalohia**
…dropdown

### What does this implement/fix? Explain your changes.

...

### Does this close any currently open issues?

...

### Any other comments?

...

### Dependent PRs/Commits

...

### Describe breaking changes, if any.

...

### Checklist

Check all those that are applicable and complete.

- [ ] Merged with latest `master` branch

### CODE REVIEW

File: core/components/atoms/dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -598,12 +617,28 @@ export class Dropdown extends React.Component<DropdownProps, DropdownState> {
...
+    if (onPopperToggle && _isOpenControlled(this.props.open)) {
+      onPopperToggle(false, type);
+    }
+  }
+
   onCancelOptions = () => {
     const { previousSelected, tempSelected, optionsLength } = this.state;
     const { selected, onUpdate } = this.props;
 
+    const popperIsOpen = _isOpenControlled(this.props.open) ? this.state.open : false;
```
Comment (aditya-kumawat) [inline core/components/atoms/dropdown/Dropdown.tsx:636]: const updatedOpen = false;

File: core/components/atoms/dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -598,12 +617,28 @@ export class Dropdown extends React.Component<DropdownProps, DropdownState> {
...
+    const values = previousSelected.map(option => option.value);
+
     if (_isControlled(selected)) {
       if (onUpdate) onUpdate('cancel-selected', previousSelected, tempSelected);
+      this.onTogglePopper(popperIsOpen, values, 'cancelClick');
```
Comment (aditya-kumawat) [inline core/components/atoms/dropdown/Dropdown.tsx:641]: this.onTogglePopper(updatedOpen, values, 'cancelClick');

File: core/components/atoms/dropdown/Dropdown.tsx [hunk 3]
```diff
@@ -613,13 +648,10 @@ export class Dropdown extends React.Component<DropdownProps, DropdownState> {
...
       triggerLabel: label,
-      open: false,
+      open: popperIsOpen,
     });
 
-    if (this.props.onClose) {
-      const values = previousSelected.map(option => option.value);
-      this.props.onClose(values, name);
-    }
+    this.onTogglePopper(popperIsOpen, values, 'cancelClick');
```
Comment (aditya-kumawat) [inline core/components/atoms/dropdown/Dropdown.tsx]: this.onTogglePopper(updatedOpen, values, 'cancelClick');

File: core/components/atoms/dropdown/Dropdown.tsx [hunk 4]
```diff
@@ -613,13 +648,10 @@ export class Dropdown extends React.Component<DropdownProps, DropdownState> {
       tempSelected: previousSelected,
       selectAll: getSelectAll(previousSelected, optionsLength),
       triggerLabel: label,
-      open: false,
+      open: popperIsOpen,
```
Comment (aditya-kumawat) [inline core/components/atoms/dropdown/Dropdown.tsx:652]: open: _isOpenControlled(this.props.open) ? this.state.open : updatedOpen

**Review by aditya-kumawat — APPROVED** [review]

**Review by satyamyadav — APPROVED** [review]

**Pull request merged**